frontend/src/router/pages/SubmitDates.jsx

```diff
@@ -68,18 +92,39 @@ function SubmitDates() {
 
     const response = await sendData(payload);
 
+    if (savingDraft) {
+      setNotes("");
+      fetchData();
+      openFlashMessage(
+        "Dates saved as draft",
+        `${season?.park.name} ${season?.featureType.name} ${season?.operatingYear} season details saved`,
+      );
+    }
+
     return response;
   }
 
-  // are there changes to save?
-  function hasChanges() {
-    const datesChanged = Object.values(dates).some((dateType) =>
-      dateType.Operation.concat(dateType.Reservation).some(
-        (dateRange) => dateRange.changed,
-      ),
-    );
+  async function submitChanges(savingDraft = false) {
+    setFormSubmitted(true);
 
-    return datesChanged || notes;
+    // Validate form state before saving
+    if (!validateForm()) {
+      console.error("Form validation failed!", errors);
+      throw new Error("Form validation failed");
+    }
+
+    if (["under review", "approved", "published"].includes(season.status)) {
+      openConfirmation(
+        "Move back to draft?",
+        "The dates will be moved back to draft and need to be submitted again to be reviewed.",
+        () => {
+          saveChanges(savingDraft);
+        },
+        "If dates have already been published, they will not be updated until new dates are submitted, approved, and published.",
+      );
+    } else {
```

Comment:
I edited some dates and clicked the "Save draft" button and this confirmation popped up, but it saved at the same time (the /save request showed in the network tab, and the success flash message came up before I clicked cancel or confirm)

Reply:
I changed the logic for showing the modal everywhere else but here. Anyway, just pushed the fix.